### ISSUE TITLE
[a11y]: ContainedList [usage examples] - additional unnecessary tab stop

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13949]
### Package

@carbon/react

### Browser
FF

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

JAWS

### Description

Option button menu  which is available after the list item

Additional tab stop after expanding the menu options

### WCAG 2.1 Violation
2.4.3 Focus Order 

### Reproduction/example

Record `Number 40` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce
After expanding when we use tab key to go to the next interactive element, there is an additional tab stop where sr is reading as "list item option contest menu".

### Recommended fix
Avoid additional tab stop. 

Note: Works as expected in the following ARIA example page which can be referred. https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-actions/

When elements receive unnecessary tab stops, keyboard navigation becomes tedious. This particularly impacts keyboard-only users with motor disabilities.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by ljcarot** [comment]
@andreancardona Can you move the content from Automated testing to the WCAG violation field?  e.g Automated testing tool and ruleset shows 2.4.3 Focus Order, but that is the WCAG violation
